Build failing on Ubuntu 18.04, gcc-9 and C++17

```
[ 92%] Built target libdbcppp_ExampleBasicUsage
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp: In function 'int main(int, char**)':
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:66:39: error: expected unqualified-id before '<' token
   66 |         catch (const boost::wrapexcept<boost::program_options::required_option>& e)
      |                                       ^
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:66:39: error: expected ')' before '<' token
   66 |         catch (const boost::wrapexcept<boost::program_options::required_option>& e)
      |               ~                       ^
      |                                       )
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:66:39: error: expected '{' before '<' token
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:66:39: error: expected primary-expression before '<' token
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:66:79: error: expected primary-expression before '>' token
   66 |         catch (const boost::wrapexcept<boost::program_options::required_option>& e)
      |                                                                               ^
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:66:82: error: 'e' was not declared in this scope
   66 |         catch (const boost::wrapexcept<boost::program_options::required_option>& e)
      |                                                                                  ^
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:112:39: error: expected unqualified-id before '<' token
  112 |         catch (const boost::wrapexcept<boost::program_options::required_option>& e)
      |                                       ^
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:112:39: error: expected ')' before '<' token
  112 |         catch (const boost::wrapexcept<boost::program_options::required_option>& e)
      |               ~                       ^
      |                                       )
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:112:39: error: expected '{' before '<' token
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:112:39: error: expected primary-expression before '<' token
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:112:79: error: expected primary-expression before '>' token
  112 |         catch (const boost::wrapexcept<boost::program_options::required_option>& e)
      |                                                                               ^
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/dbcppp/main.cpp:112:82: error: 'e' was not declared in this scope
  112 |         catch (const boost::wrapexcept<boost::program_options::required_option>& e)
      |                                                                                  ^
src/dbcppp/CMakeFiles/dbcppp.dir/build.make:82: recipe for target 'src/dbcppp/CMakeFiles/dbcppp.dir/main.cpp.o' failed
make[2]: *** [src/dbcppp/CMakeFiles/dbcppp.dir/main.cpp.o] Error 1
CMakeFiles/Makefile2:240: recipe for target 'src/dbcppp/CMakeFiles/dbcppp.dir/all' failed
make[1]: *** [src/dbcppp/CMakeFiles/dbcppp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 96%] Linking CXX executable ../../bin/libdbcppp_Test
[ 96%] Built target libdbcppp_Test
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2

```

I was using 1.65. I've upgraded to 1.73 through a ppa, and now the build fails like this:

```
Scanning dependencies of target libdbcppp
[ 12%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/AttributeDefinitionImpl.cpp.o
[ 12%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/AttributeImpl.cpp.o
[ 12%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/BitTimingImpl.cpp.o
[ 16%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/CApi.cpp.o
[ 20%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/DBCAST2Network.cpp.o
[ 28%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/Network2C.cpp.o
[ 28%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/KCD2Network.cpp.o
[ 32%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/Network2DBC.cpp.o
[ 36%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/MessageImpl.cpp.o
[ 40%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/NetworkImpl.cpp.o
[ 52%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/EnvironmentVariableImpl.cpp.o
[ 56%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/SignalTypeImpl.cpp.o
[ 52%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/SignalImpl.cpp.o
[ 52%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/ValueTableImpl.cpp.o
[ 60%] Building CXX object src/libdbcppp/CMakeFiles/libdbcppp.dir/NodeImpl.cpp.o
In file included from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.h:4,
                 from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.cpp:3:
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/AttributeDefinition.h:50:21: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   50 |         static std::unique_ptr<AttributeDefinition> create(
      |                     ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/AttributeDefinition.h:11:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   10 | #include "Export.h"
  +++ |+#include <memory>
   11 | 
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/AttributeDefinition.h:55:22: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   55 |         virtual std::unique_ptr<AttributeDefinition> clone() const = 0;
      |                      ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/AttributeDefinition.h:55:17: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   55 |         virtual std::unique_ptr<AttributeDefinition> clone() const = 0;
      |                 ^~~
In file included from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.cpp:3:
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.h:14:22: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   14 |         virtual std::unique_ptr<AttributeDefinition> clone() const override;
      |                      ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.h:5:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
    4 | #include "../../include/dbcppp/AttributeDefinition.h"
  +++ |+#include <memory>
    5 | 
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.cpp:7:6: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
    7 | std::unique_ptr<AttributeDefinition> AttributeDefinition::create(
      |      ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.cpp:4:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
    3 | #include "AttributeDefinitionImpl.h"
  +++ |+#include <memory>
    4 | 
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.cpp:23:6: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   23 | std::unique_ptr<AttributeDefinition> AttributeDefinitionImpl::clone() const
      |      ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeDefinitionImpl.cpp:23:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   23 | std::unique_ptr<AttributeDefinition> AttributeDefinitionImpl::clone() const
      | ^~~
src/libdbcppp/CMakeFiles/libdbcppp.dir/build.make:82: recipe for target 'src/libdbcppp/CMakeFiles/libdbcppp.dir/AttributeDefinitionImpl.cpp.o' failed
make[2]: *** [src/libdbcppp/CMakeFiles/libdbcppp.dir/AttributeDefinitionImpl.cpp.o] Error 1
make[2]: *** Se espera a que terminen otras tareas....
In file included from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/Attribute.h:9,
                 from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.h:5,
                 from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:3:
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/AttributeDefinition.h:50:21: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   50 |         static std::unique_ptr<AttributeDefinition> create(
      |                     ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/AttributeDefinition.h:1:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
  +++ |+#include <memory>
    1 | 
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/AttributeDefinition.h:55:22: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   55 |         virtual std::unique_ptr<AttributeDefinition> clone() const = 0;
      |                      ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/AttributeDefinition.h:55:17: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   55 |         virtual std::unique_ptr<AttributeDefinition> clone() const = 0;
      |                 ^~~
In file included from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.h:5,
                 from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:3:
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/Attribute.h:20:21: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   20 |         static std::unique_ptr<Attribute> create(
      |                     ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/Attribute.h:10:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
    9 | #include "AttributeDefinition.h"
  +++ |+#include <memory>
   10 | 
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/Attribute.h:25:22: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   25 |         virtual std::unique_ptr<Attribute> clone() const = 0;
      |                      ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/Attribute.h:25:17: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   25 |         virtual std::unique_ptr<Attribute> clone() const = 0;
      |                 ^~~
In file included from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:3:
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.h:15:22: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
   15 |         virtual std::unique_ptr<Attribute> clone() const override;
      |                      ^~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.h:6:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
    5 | #include "../../include/dbcppp/Attribute.h"
  +++ |+#include <memory>
    6 | 
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:8:28: error: no declaration matches ‘std::unique_ptr<dbcppp::Attribute> dbcppp::Attribute::create(std::string&&, dbcppp::AttributeDefinition::ObjectType, dbcppp::Attribute::value_t)’
    8 | std::unique_ptr<Attribute> Attribute::create(
      |                            ^~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:8:28: note: no functions named ‘std::unique_ptr<dbcppp::Attribute> dbcppp::Attribute::create(std::string&&, dbcppp::AttributeDefinition::ObjectType, dbcppp::Attribute::value_t)’
In file included from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.h:5,
                 from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:3:
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/../../include/dbcppp/Attribute.h:14:22: note: ‘class dbcppp::Attribute’ defined here
   14 |     class DBCPPP_API Attribute
      |                      ^~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:24:28: error: no declaration matches ‘std::unique_ptr<dbcppp::Attribute> dbcppp::AttributeImpl::clone() const’
   24 | std::unique_ptr<Attribute> AttributeImpl::clone() const
      |                            ^~~~~~~~~~~~~
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:24:28: note: no functions named ‘std::unique_ptr<dbcppp::Attribute> dbcppp::AttributeImpl::clone() const’
In file included from /home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.cpp:3:
/home/david/CTAG/repos/ml_logging_platform/dependencies/dbcppp/src/libdbcppp/AttributeImpl.h:9:11: note: ‘class dbcppp::AttributeImpl’ defined here
    9 |     class AttributeImpl final
      |           ^~~~~~~~~~~~~
src/libdbcppp/CMakeFiles/libdbcppp.dir/build.make:95: recipe for target 'src/libdbcppp/CMakeFiles/libdbcppp.dir/AttributeImpl.cpp.o' failed
make[2]: *** [src/libdbcppp/CMakeFiles/libdbcppp.dir/AttributeImpl.cpp.o] Error 1
CMakeFiles/Makefile2:213: recipe for target 'src/libdbcppp/CMakeFiles/libdbcppp.dir/all' failed
make[1]: *** [src/libdbcppp/CMakeFiles/libdbcppp.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,8 @@
-﻿
+
 cmake_minimum_required (VERSION 3.8)
 
 project("libdbcppp")
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_STATIC_LIBRARY_PREFIX "")
 set(CMAKE_SHARED_LIBRARY_PREFIX "")
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)

--- a/include/dbcppp/AttributeDefinition.h
+++ b/include/dbcppp/AttributeDefinition.h
@@ -6,6 +6,7 @@
 #include <vector>
 #include <functional>
 #include <boost/variant.hpp>
+#include <memory>
 
 #include "Export.h"
 

